lib/screens/forgot_password/enter_otp.dart

```diff
@@ -0,0 +1,297 @@
+                            ),
+                          ),
+                          Expanded(
+                            flex: 1,
+                            child: Icon(Icons.verified_user,
+                                color: Colors.white, size: 65 * heightFactor),
+                          ),
+                          Expanded(
+                            flex: 1,
+                            child: Container(
+                              alignment: Alignment.center,
+                              child: Text(
+                                "Verify OTP",
+                                style: TextStyle(
+                                  fontWeight: FontWeight.w300,
+                                  color: C.primaryHighlightedColor,
+                                  fontSize: heightFactor * 30,
+                                ),
+                              ),
+                            ),
+                          ),
+                          Container(
+                            padding:
+                                EdgeInsets.fromLTRB(heightFactor * 10, 0, heightFactor * 10, 0),
+                            alignment: Alignment.center,
+                            child: Column(
+                              children: [
+                                Text(
+                                  "An otp has been sent to the registered user",
+                                  style: TextStyle(
+                                    fontSize: heightFactor * 20,
+                                  ),
+                                ),
+                                Padding(
+                                    padding: EdgeInsets.only(top: 10 * heightFactor),
+                                    child: Text(
+                                      "email",
+                                      style: TextStyle(
+                                        fontSize: heightFactor * 20,
+                                      ),
+                                    )),
+                              ],
+                            ),
+                          ),
+                        ],
+                      ),
+                    ),
+                  ),
+                  Flexible(
+                    flex: 3,
+                    child: Column(
+                      children: [
+                        //!OTP number fields
+                        Padding(
+                          padding: const EdgeInsets.only(top: 20),
+                          child: Row(
+                            mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                            children: [
+                              Container(
+                                child: Padding(
+                                  padding: const EdgeInsets.all(16.0),
+                                  child: Text(
+                                    otp[0] ?? "",
```

Comment:
Exception caught: `RangeError (index): Invalid value: Valid value range is empty: 0`
Meaning: you tried to get an element at position 0, but there was no element there since string is empty.
What you did wrong: when calling `otp[0] ?? ""`, you mean to say, if element at position 0 is null, return `""`. But in this case, the element is not only not null, it doesn't even exist.
Fix it.